src/queueUntilType.js

```diff
@@ -0,0 +1,35 @@
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+export function queueUntilType(expectedType) {
+  return this.lift(function(source) {
+    let queue = [];
+    let hasSeenType = false;
+    const nullifyQueue = () => { queue = null; };
+
+    const outer = {
+      notifyNext: (_, action) => this.next(action),
+      notifyComplete: nullifyQueue
+    };
+    return source.subscribe({
+      next: action => {
+        // It's important to check this first because of possible re-entry
+        // before we've finished flushing the queue.
+        if (hasSeenType) {
+          return this.next(action);
+        }
+
+        if (action && action.type === expectedType) {
+          hasSeenType = true;
+          this.add(subscribeToResult(outer, queue));
+          // no longer needed, free up memory
+          nullifyQueue();
+        } else {
+          queue.push(action);
+        }
+      },
+      error: e => this.error(e),
+      complete: () => this.complete(),
+      unsubscribe: nullifyQueue
```

Comment:
> I'm just indeed very curious!

Yes now I am curious as well. My intuition tells me this is somehow a consequence of the lift pattern. That the unsubscribe logic must be bundled with the observer (or what is now internally called the subscriber).

Yes the unsubscribe is called in the above code:

<img width="634" alt="hyper" src="https://user-images.githubusercontent.com/6775919/27249207-c89e9ef4-52c4-11e7-8268-3dd34493b9c6.png">

<img width="724" alt="hyper" src="https://user-images.githubusercontent.com/6775919/27249220-106058a4-52c5-11e7-9a7e-315c7df590f6.png">

It is strange that in your finally example code above it is not called. I believe that is most likely a bug.